code/modules/mob/living/carbon/human/death.dm

```diff
@@ -47,12 +47,18 @@
 		ticker.mode.update_cult_icons_removed(mind)
 		ticker.mode.cult -= mind
 
-	species = null
+	if(species)
+		qdel(species)
+		species = null
```

Comment:
I'm pretty sure every mob has its own species object

However, external limbs use the species datums too, so I wouldn't delete them